src/plugins/data/public/ui/saved_query_management/saved_query_management_component.tsx

```diff
@@ -62,14 +62,25 @@ export const SavedQueryManagementComponent: FunctionComponent<Props> = ({
   const [savedQueries, setSavedQueries] = useState([] as SavedQuery[]);
   const [count, setTotalCount] = useState(0);
   const [activePage, setActivePage] = useState(0);
+  const cancelPendingListingRequest = useRef<() => void>(() => {});
 
   useEffect(() => {
     const fetchCountAndSavedQueries = async () => {
-      const savedQueryCount = await savedQueryService.getSavedQueryCount();
-      setTotalCount(savedQueryCount);
+      cancelPendingListingRequest.current();
+      let requestGotCancelled = false;
+      cancelPendingListingRequest.current = () => {
+        requestGotCancelled = true;
+      };
+
+      const [savedQueryCount, savedQueryItems] = await Promise.all([
+        savedQueryService.getSavedQueryCount(),
```

Comment:
Why do we need `getSavedQueryCount` in the first place?
Couldn't we just fetch the saved queries and then their count is `sortedSavedQueryItems.length`?

Reply:
I think that's not necessarily the same number - `findSavedQueries` just fetches a single page, but there could be (much) more queries.

Not sure how that would fix the problem - the pending listing request and the delete request can still arrive at different times and cause inconsistent state in the client:

* User opens popover (with a cached version of the current list)
* This triggers a request for the listing which runs in the background, but the user can already interact because there is already a list of queries from the last time the popover was open
* The user deletes a query (listing request still pending) - the delete request is sent to the server
* The local cache of queries is modified to remove the saved query in https://github.com/elastic/kibana/pull/58433/files#diff-1eef7731b4c8696b004eee34196e6106L106
* Now the listing request arrives, but it is the state before the removal. The local cache gets updated with the state before the deletion (which is probably still in flight)
* The user sees the query they just deleted and wonders whether the deletion actually worked (closing and opening the popover would fix the problem because it fetches the listing again, but the user and our functional test doesn't know this)

Reply:
The response for the saved_objects "Saved Queries" request, does contain the total number of "Saved Queries". Maybe we could change the API of `findSavedQueries` to return the list and the total?

Reply:
> The response for the saved_objects "Saved Queries" request, does contain the total number of "Saved Queries". Maybe we could change the return value of findSavedQueries to return the list and the total?

Makes sense to me, I will look into this since this component is the only consumer anyway.

> Also, it seems like the pagination doesn't work now:
> I set my page size to 5 and have 7 saved queries, and I can't get to them from the UI

What did you change? It works fine for me on this PR with 
```ts
const perPage = 3;
```

<img width="414" alt="Screenshot 2020-02-25 at 18 29 52" src="https://user-images.githubusercontent.com/1508364/75271668-d9763800-57fc-11ea-9c97-ced1b5e3a73a.png">
<img width="405" alt="Screenshot 2020-02-25 at 18 29 57" src="https://user-images.githubusercontent.com/1508364/75271671-daa76500-57fc-11ea-9dea-0c675e273ca2.png">

Reply:
Might have changed the wrong thing. Will retest once the PR is updated.